nova_vm/src/ecmascript/builtins/control_abstraction_objects/promise_objects/promise_constructor.rs

```diff
@@ -102,20 +174,34 @@ impl PromiseConstructor {
     fn race(_agent: &mut Agent, _this_value: Value, _arguments: ArgumentsList) -> JsResult<Value> {
         todo!()
     }
-    fn reject(
-        _agent: &mut Agent,
-        _this_value: Value,
-        _arguments: ArgumentsList,
-    ) -> JsResult<Value> {
-        todo!()
+
+    fn reject(agent: &mut Agent, _this_value: Value, arguments: ArgumentsList) -> JsResult<Value> {
+        // 1. Let C be the this value.
+        // 2. Let promiseCapability be ? NewPromiseCapability(C).
```

Comment:
**suggestion:** Should we maybe intentionally crash on Promise subclassing?

Reply:
Fixed the constructor, `reject` and `resolve` to do so.